### PR TITLE
Automation of Service bindings | dev-console

### PR DESCRIPTION
**Pull request by SparshKesari**
**Description:**

- Automation of Create workload from Operator Backed file (create-operator-backed.feature)
- Automation of Provide quick search in Add page (quick-search-add.feature)

Functionality PR : 
- https://github.com/openshift/console/pull/11313 (merged)

Gherkin Script PR :
-  https://github.com/openshift/console/pull/11326 (merged)

**Jira Id:** [ODC-6513](https://issues.redhat.com/browse/ODC-6513)

**Pre-conditions/setup:**

**Checks for approving Epic scenarios Automation PR:**

- [x] Execute the @to-do tagged gherkin scripts manually
- [x] Convert the @to-do gherkin scripts to cypress automation scripts
- [x] Once scripts are automated, replace tag @to-do with @epic-number
- [x] Execute the scripts in Remote cluster

**Refactoring criteria for approving Automation PR:**

- [ ] update the test cases count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)

**Execution Commands:**
Example: 
```
export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
    BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
    BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.8-042801.devcluster.openshift.com
    export BRIDGE_KUBEADMIN_PASSWORD
    export BRIDGE_BASE_ADDRESS
    oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD
    oc apply -f ./frontend/integration-tests/data/htpasswd-secret.yaml
    oc patch oauths cluster --patch "$(cat ./frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
    ./test-cypress.sh -p dev-console

```

**Screen shots**:

quick-search-add.feature
image

create-operator-backed.feature

image

**Browser conformance**:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-operator-backed.feature [hunk 1]
```diff
@@ -45,18 +45,18 @@ Feature: Create workload from Operator Backed file
...
+        @regression @odc-6467
         Scenario: Bindable resource in Operator Backed: A-08-TC05
             Given user has installed Service Binding operator
               And user has installed Crunchy Postgres for Kubernetes operator
               And user has created or selected namespace "aut-service-binding"
               And user is at OperatorBacked page
              When user enters "Postgres Cluster" in Filter by keyword
-             Then user will see "Bindable" label associated with "Postgres Cluster" card
-              And user will see "Bindable" label in "Postgres Cluster" sidebar
+             Then user will see Bindable label associated with catalog card
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-operator-backed.feature:55]: Why is the Bindable label removed from the quotes, you can use it with `contains`  using variable

File: frontend/packages/dev-console/integration-tests/support/step-definitions/addFlow/quick-search-add.ts [hunk 2]
```diff
@@ -80,3 +82,15 @@ Then(
...
+Given('user has installed Service Binding operator', () => {
+  verifyAndInstallOperator(operators.ServiceBinding);
+});
+
+Given('user has installed Crunchy Postgres for Kubernetes operator', () => {
+  verifyAndInstallOperator(operators.CrunchyPostgresforKubernetes);
+});
+
+Then('user will see Bindable label associated with {string}', () => {
+  cy.get(catalogPO.catalogBatch).should('contain.text', 'Bindable');
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/support/step-definitions/addFlow/quick-search-add.ts:95]: Where is the variable for the string

File: frontend/packages/dev-console/integration-tests/features/addFlow/quick-search-add.feature [hunk 3]
```diff
@@ -57,11 +57,12 @@ Feature: Provide quick search in Add page
...
         Scenario: Bindable resource in Quick Add: A-11-TC07
             Given user has installed Service Binding operator
-              And user has installed Crunchy Postgres for Kubernetes operator
+              And user is at developer perspective
+            Given user has installed Crunchy Postgres for Kubernetes operator
               And user is at Add page
              When user clicks Add to project button
               And user enters "crunchy" in Add to project search bar
-             Then user will see "Bindable" label associated with "Postgres Cluster"
+             Then user will see Bindable label associated with "Postgres Cluster"
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/addFlow/quick-search-add.feature:68]: Use Bindable in quotes

Comment (SparshKesari) [reply frontend/packages/dev-console/integration-tests/features/addFlow/quick-search-add.feature]: @sanketpathak made the changes. 👍

File: frontend/packages/dev-console/integration-tests/features/addFlow/quick-search-add.feature [hunk 4]
```diff
@@ -57,10 +57,11 @@ Feature: Provide quick search in Add page
              Then user is taken to the Topology page with "devfile-sample-git" workload created
 
 
-        @regression @to-do @odc-6467
+        @regression @odc-6467
         Scenario: Bindable resource in Quick Add: A-11-TC07
             Given user has installed Service Binding operator
-              And user has installed Crunchy Postgres for Kubernetes operator
+              And user is at developer perspective
+            Given user has installed Crunchy Postgres for Kubernetes operator
```
Comment (jrichter1) [inline frontend/packages/dev-console/integration-tests/features/addFlow/quick-search-add.feature:64]: just a nitpick, you now have 2 `Given`s in one chain

Comment (SparshKesari) [reply frontend/packages/dev-console/integration-tests/features/addFlow/quick-search-add.feature]: Sorry, I missed it. Resolved @jrichter1

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-operator-backed.feature [hunk 5]
```diff
@@ -35,6 +36,7 @@ Feature: Create workload from Operator Backed file
         @regression @manual
         Scenario: Create the Jaeger workload from Operator Backed: A-08-TC03
             Given user has installed Jaeger operator
+              And user has installed OpenShift Serverless Operator
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-operator-backed.feature:39]: nit: You can remove it from here

**Comment by sanketpathak** [comment]
Tests are working green
/lgtm

**Comment by sanketpathak** [comment]
PR only affects E2E tests
/label qe-approved
/label docs-approved
/label px-approved

**Pull request merged**